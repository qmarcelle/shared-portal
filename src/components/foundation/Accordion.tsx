import { ReactNode, useState } from 'react';
import { IComponent } from '../IComponent';

export interface AccordionTileProps extends IComponent {
  icon?: ReactNode;
  closeIcon?: ReactNode;
  openIcon?: ReactNode;
  label: ReactNode;
  subLabel?: ReactNode;
  child: ReactNode;
  initialOpen: boolean;
  type?: 'normal' | 'card';
}

export const Accordion = ({
  className,
  icon,
  closeIcon,
  openIcon,
  label,
  subLabel,
  child,
  initialOpen,
  type = 'normal',
}: AccordionTileProps) => {
  const [open, setOpen] = useState(initialOpen);

  return (
    <div
      onClick={type == 'card' ? () => setOpen(!open) : undefined}
      className={`flex flex-col ${type == 'card' && 'card-main-elevated p-1 cursor-pointer'} ${className}`}
    >
      <div
        onClick={() => setOpen(!open)}
        className="flex flex-row m-1 accordion-head"
      >
        {icon && <div className="mr-1">{icon}</div>}
        <p className="mr-2 flex-grow">{label}</p>
        {open ? closeIcon : openIcon}
      </div>
      {!open && <div className="mr-2 flex-grow">{subLabel}</div>}
      {open && <div className="m-1">{child}</div>}
      {type == 'normal' && <div className="divider"></div>}
    </div>
  );
};
