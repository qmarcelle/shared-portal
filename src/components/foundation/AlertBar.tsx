import Image from 'next/image';
import { useState } from 'react';
import AlertIcon from '../../../public/assets/Alert-White.svg';
import CloseIcon from '../../../public/assets/Close-White.svg';

export interface AlertBarProps {
  alerts: string[];
}

const AlertTile = ({
  index,
  label,
  closeCallback,
}: {
  index: number;
  label: string;
  closeCallback: (index: number) => void;
}) => {
  return (
    <div className="flex flex-row py-1 px-2 w-full alert-tile">
      <div>
        <Image src={AlertIcon} className="icon" alt={'AlertIcon'} />
      </div>
      <p className="mx-1 grow">{label}</p>
      <div onClick={() => closeCallback(index)}>
        <Image src={CloseIcon} className="icon" alt={'CloseIcon'} />
      </div>
    </div>
  );
};

export const AlertBar = ({ alerts }: AlertBarProps) => {
  const [items, setItems] = useState(alerts);

  const closeCallback = (index: number) => {
    items.splice(index, 1);
    setItems([...items]);
  };

  if (items.length == 0) {
    return null;
  }

  if (items.length == 1) {
    return (
      <AlertTile
        key={items[0]}
        index={0}
        label={items[0]}
        closeCallback={closeCallback}
      />
    );
  } else {
    return (
      <div className="flex flex-col self-stretch">
        {items.map((item, index) => {
          if (index != alerts.length - 1) {
            return (
              <div key={index}>
                <AlertTile
                  key={item}
                  index={index}
                  label={item}
                  closeCallback={closeCallback}
                />
                <div className="divider"></div>
              </div>
            );
          } else {
            return (
              <AlertTile
                key={item}
                index={index}
                label={item}
                closeCallback={closeCallback}
              />
            );
          }
        })}
      </div>
    );
  }
};
