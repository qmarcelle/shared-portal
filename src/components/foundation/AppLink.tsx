import { ReactNode } from 'react';
import { IComponent } from '../IComponent';

export interface LinkProps extends IComponent {
  url?: string;
  label: string;
  icon?: ReactNode | null;
  displayStyle?: string;
  linkUnderline?: string;
  callback?: () => void;
}

export const AppLink = ({
  label,
  icon,
  url,
  callback,
  className,
  linkUnderline,
  displayStyle = 'block',
}: LinkProps) => {
  return (
    <a style={{ display: `${displayStyle}` }} tabIndex={-1} href={url}>
      <button
        onClick={callback}
        style={{
          maxWidth: 'max-content',
          height: 'auto',
          display: `${displayStyle}`,
        }}
        tabIndex={0}
        className={`flex flex-row link-container ${className}`}
      >
        <p
          className={`link ${linkUnderline}`}
          style={{ display: `${displayStyle}` }}
        >
          {label}
        </p>
        {icon && (
          <p className="ml-1" style={{ display: `${displayStyle}` }}>
            {icon}
          </p>
        )}
      </button>
    </a>
  );
};
