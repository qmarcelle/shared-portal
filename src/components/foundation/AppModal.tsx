import FocusTrap from 'focus-trap-react';
import Image from 'next/image';
import React, { ReactElement, createContext } from 'react';
import { Modal } from 'react-responsive-modal';
import { create } from 'zustand';
import leftIcon from '../../../public/assets/left.svg';
import { IComponent } from '../IComponent';
import { Column } from './Column';
import { bcbstSilhouletteLogo, closeIcon } from './Icons';
import { Row } from './Row';
import { Spacer } from './Spacer';
import { TextBox } from './TextBox';

/* eslint-disable @typescript-eslint/no-explicit-any */
interface ModalProps extends IComponent {
  content: ReactElement & ModalChildProps;
  store?: any;
  isChildActionAppModal?: boolean;
  childModalContent?: ReactElement & ModalChildProps;
}
interface ModalControllerProps {
  showBack: boolean;
  showModal: boolean;
  pageIndex: number;
  store: null;
  isChildAction?: boolean;
  updateShowBack: (isShowBack: boolean) => void;
  updateshowModal: (isShowBack: boolean) => void;
  updatepageIndex: (pageIndex: number) => void;
  updateStore: (store: null) => void;
  showAppModal: (props: ModalProps) => void;
  dismissModal: () => void;
  isChildModal: boolean;
  showChildModal: () => void;
  dismissChildModal: () => void;
  isChildActionAppModalProp: boolean;
}

export const useAppModalStore = create<ModalControllerProps>((set) => ({
  showBack: false,
  showModal: false,
  isChildModal: false,
  pageIndex: 0,
  store: null,
  isChildActionAppModalProp: false,
  updateShowBack: (isShowBack: boolean) =>
    set(() => ({ showBack: isShowBack })),
  updateshowModal: (isShowModal: boolean) =>
    set(() => ({ showModal: isShowModal })),
  updatepageIndex: (pageIndex: number) => set(() => ({ pageIndex: pageIndex })),
  updateStore: (store: null) => set(() => ({ store: store })),
  showAppModal: ({
    content,
    store,
    isChildActionAppModal,
    childModalContent,
  }: ModalProps) => {
    modalBody = content;
    childModalBody = childModalContent;
    if (isChildActionAppModal) {
      set(() => ({
        isChildActionAppModalProp: isChildActionAppModal,
      }));
    }
    set(() => ({
      showModal: true,
    }));
    set(() => ({
      store: store,
    }));
    pageIndexStack = [0];
  },
  dismissModal: () => {
    set(() => ({
      showModal: false,
    }));
    set(() => ({
      store: null,
    }));
    set(() => ({
      showBack: false,
    }));
    set(() => ({
      pageIndex: 0,
    }));
    set(() => ({
      isChildModal: false,
    }));
    modalBody = null;
  },
  showChildModal: () => {
    set(() => ({
      isChildModal: true,
    }));
  },
  dismissChildModal: () => {
    set(() => ({
      isChildModal: false,
    }));
  },
}));
let modalBody: ReactElement | null = null;
let childModalBody: ReactElement | null | undefined = null;
let pageIndexStack: number[] = [];
export interface ModalChildProps {
  changePage?: (pageIndex: number, showBackButton?: boolean) => any;
  pageIndex?: number;
}

export const ModalContext = createContext<any>(null);

interface ModalHeaderProps {
  onClose: () => any;
}

const ModalHeader = ({ onClose }: ModalHeaderProps) => {
  const { updatepageIndex, updateShowBack, pageIndex, showBack } =
    useAppModalStore();
  const onBackPressed = () => {
    pageIndexStack.pop();
    const popIndex = pageIndexStack[pageIndexStack.length - 1];
    updatepageIndex(popIndex ?? 0);
    if (pageIndex == 0) {
      updateShowBack(false);
    }
  };

  const handleKeyDown = (e: {
    preventDefault: () => void;
    keyCode: number;
  }) => {
    e.preventDefault();
    if (e.keyCode === 13) {
      onClose();
    }
  };

  return (
    <Row className="justify-between modal-header items-center relative">
      {showBack ? (
        <Row
          tabIndex={1}
          onClick={onBackPressed}
          className="items-center focus-icon"
        >
          <Image src={leftIcon} className="modal-icon" alt="back" />
          <TextBox text="Back" className="primary-color underline" />
        </Row>
      ) : (
        <div tabIndex={1} />
      )}

      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon modal-header-logo absolute m-auto left-0 right-0"
        alt="bcbst logo"
      />
      <div
        tabIndex={0}
        className="size-8 focus-icon"
        onKeyDown={handleKeyDown}
        onClick={onClose}
      >
        <Image src={closeIcon} className="size-4" alt="close" />
      </div>
    </Row>
  );
};

const ModalFooter = () => {
  return (
    <Row className="justify-center modal-footer">
      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon"
        alt="bcbst logo"
      />
    </Row>
  );
};

export const AppModal = () => {
  const {
    dismissModal,
    updateShowBack,
    updatepageIndex,
    store,
    showModal,
    pageIndex,
    isChildModal,
    showChildModal,
    dismissChildModal,
    isChildActionAppModalProp,
  } = useAppModalStore();
  const changePage = (pageIndex: number, showBackButton: boolean = false) => {
    updateShowBack(showBackButton);
    updatepageIndex(pageIndex);
    pageIndexStack.push(pageIndex);
  };
  const closeModal = () => {
    isChildActionAppModalProp ? showChildModal() : dismissModal();
  };

  const closeChildModal = () => {
    dismissChildModal();
  };
  const closeAllModal = () => {
    dismissModal();
  };
  return (
    <ModalContext.Provider value={store}>
      <InnerAppModal
        showModal={showModal}
        modalBody={modalBody ?? <></>}
        changePage={changePage}
        pageIndex={pageIndex}
        closeModal={closeModal}
        isChildModal={isChildModal}
        closeChildModal={closeChildModal}
        closeAllModal={closeAllModal}
        childModalBody={childModalBody ?? <></>}
      />
    </ModalContext.Provider>
  );
};

type InnerAppModalProps = {
  showModal?: boolean;
  closeModal?: () => any;
  modalBody: ReactElement;
  changePage?: (pageIndex: number, showBackButton: boolean) => any;
  pageIndex?: number;
  isChildModal?: boolean;
  closeChildModal?: () => any;
  closeAllModal?: () => any;
  childModalBody?: ReactElement;
};

export const InnerAppModal = ({
  showModal = true,
  closeModal = () => {},
  modalBody,
  changePage = () => {},
  pageIndex = 0,
  isChildModal = false,
  childModalBody,
}: InnerAppModalProps) => {
  return (
    <Modal
      classNames={{
        overlay: 'modal-overlay',
      }}
      showCloseIcon={false}
      open={showModal}
      closeOnOverlayClick={true}
      onClose={closeModal}
    >
      <FocusTrap>
        <div>
          <Column className="modal-container">
            <ModalHeader onClose={closeModal} />
            <Spacer size={32} />
            <div className="flex flex-row flex-grow justify-center">
              <div className="flex flex-row justify-center modal-content">
                {modalBody ? (
                  React.cloneElement(modalBody, {
                    changePage,
                    pageIndex: pageIndex,
                  })
                ) : (
                  <div />
                )}
              </div>
            </div>
            {isChildModal && (
              <div className="child-modal">
                <div className="child-modal-content">{childModalBody}</div>
              </div>
            )}
            <Spacer size={32} />
            <ModalFooter />
          </Column>
        </div>
      </FocusTrap>
    </Modal>
  );
};
