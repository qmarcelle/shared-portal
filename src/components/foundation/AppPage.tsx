import { ReactNode } from 'react';
import { IComponent } from '../IComponent';

interface PageProps extends IComponent {
  children: ReactNode;
}

export const AppPage = ({
  className = '',
  children,
  onClick,
  tabIndex,
}: PageProps) => {
  return (
    <main
      tabIndex={tabIndex}
      className={`flex flex-col justify-center items-center page ${className}`.trimEnd()}
      onClick={onClick}
    >
      {children}
    </main>
  );
};
