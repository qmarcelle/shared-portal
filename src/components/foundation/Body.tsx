import { ReactNode } from 'react';
import { IComponent } from '../IComponent';
import { Column } from './Column';

interface PageProps extends IComponent {
  children: ReactNode;
}

export const Body = ({
  className = '',
  children,
  onClick,
  tabIndex,
}: PageProps) => {
  return (
    <Column
      tabIndex={tabIndex}
      className={`app-content app-base-font-color ${className}`.trimEnd()}
      onClick={onClick}
    >
      {children}
    </Column>
  );
};
