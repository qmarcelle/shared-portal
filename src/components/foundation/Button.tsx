import { ReactNode } from 'react';
import { IComponent } from '../IComponent';
type ButtonType =
  | 'primary'
  | 'secondary'
  | 'ghost'
  | 'rounded'
  | 'pill'
  | 'elevated'
  | 'card';

interface ButtonProps extends IComponent {
  type?: ButtonType;
  id?: string;
  label?: string;
  icon?: ReactNode;
  callback?: () => void | Promise<void> | null;
}

export const Button = ({
  type = 'primary',
  label,
  icon,
  id,
  className,
  callback,
}: ButtonProps) => {
  function computeContent() {
    if (label) {
      return (
        <>
          {label}
          {icon && <p className="ml-2">{icon}</p>}
        </>
      );
    } else {
      return icon;
    }
  }

  return (
    <button
      onClick={callback}
      aria-label={label}
      className={`button-text ${type} text-center flex flex-row justify-center items-center min-w-fit ${callback == null ? 'inactive' : null} ${className}`}
      type="button"
      id={id}
    >
      {computeContent()}
    </button>
  );
};
