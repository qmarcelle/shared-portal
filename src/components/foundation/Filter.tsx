import Image from 'next/image';
import { useState } from 'react';
import downIcon from '../../../public/assets/down.svg';
import resetIcon from '../../../public/assets/reset.svg';
import {
  FilterDetails,
  FilterItem,
} from '../../models/filter_dropdown_details';
import { IComponent } from '../IComponent';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Header } from '../foundation/Header';
import { RichDropDown } from '../foundation/RichDropDown';
import { Row } from '../foundation/Row';
import { Spacer } from '../foundation/Spacer';
import { TextBox } from '../foundation/TextBox';
import { Button } from './Button';
import { TextField } from './TextField';

interface FilterProps extends IComponent {
  filterHeading: string;
  filterItems: FilterItem[];
  buttons?: {
    type: 'primary';
    className: string;
    label: string;
    callback: (isClicked: boolean) => void | Promise<void> | null;
  };
}

const FilterTile = ({ user }: { user: FilterDetails }) => {
  return (
    <Column className="border-none flex-grow">
      <TextBox type="body-1" text={`${user.label}`} />
    </Column>
  );
};

const FilterHead = ({ user }: { user: FilterDetails }) => {
  return (
    <div className="body-1 input">
      <Row className="p-1 items-center">
        <FilterTile user={user} />
        <Image
          src={downIcon}
          className="w-[20px] h-[20px] ml-2 items-end"
          alt=""
        />
      </Row>
    </div>
  );
};

export const Filter = ({
  className,
  filterHeading,
  filterItems,
  buttons,
}: FilterProps) => {
  const [reset, resetFilter] = useState(false);
  const [filterItem, setFilterItem] = useState(filterItems);

  const handleReset = () => {
    resetFilter(false);
    setFilterItem(filterItems);
    buttons?.callback(false);
  };

  const handleDropDownUpdate = (value: FilterDetails, index: number) => {
    const dropdDownCopiedVal = JSON.parse(JSON.stringify(filterItem));
    if (dropdDownCopiedVal[index]) {
      dropdDownCopiedVal[index].selectedValue = value;
    }
    setFilterItem(dropdDownCopiedVal);
    resetFilter(true);
  };

  const handleInputUpdate = (value: string, index: number) => {
    const filterList = JSON.parse(JSON.stringify(filterItem));
    if (filterList[index]) {
      filterList[index].value = value;
    }
    setFilterItem(filterList);
    resetFilter(true);
  };

  const handleCallback = () => {
    buttons?.callback(true);
  };

  return (
    <Card className={className}>
      <Column>
        <Header className="title-2" text={filterHeading} />
        <Spacer size={32} />
        {filterItem.slice(0, filterItem.length).map((item, index) => (
          <>
            <Column key={index}>
              {item.type == 'dropdown' ? (
                <div className="body-1">{item.label} </div>
              ) : null}
              <Spacer size={5} />
              {item.type == 'dropdown' ? (
                <div>
                  <RichDropDown<FilterDetails>
                    headBuilder={(val) => <FilterHead user={val} />}
                    itemData={item.value as FilterDetails[]}
                    itemsBuilder={(data, index) => (
                      <FilterTile user={data} key={index} />
                    )}
                    selected={item.selectedValue ?? ({} as FilterDetails)}
                    onSelectItem={(val) => {
                      handleDropDownUpdate(val, index);
                    }}
                  />
                </div>
              ) : null}
              {item.type == 'input' ? (
                <TextField
                  type="text"
                  label={item.label}
                  value={item.value as string}
                  valueCallback={(value) => {
                    handleInputUpdate(value, index);
                  }}
                ></TextField>
              ) : null}
            </Column>
            <Spacer size={16} />
          </>
        ))}
        <Spacer size={16} />
        {reset && (
          <a className="link flex !no-underline" href="#" onClick={handleReset}>
            <Image
              src={resetIcon}
              className="w-[20px] h-[20px] ml-2 mr-2 items-end"
              alt=""
            />
            Reset Filter
          </a>
        )}
        <Spacer size={16} />
        {buttons ? (
          <Button
            className={buttons.className}
            label={buttons.label}
            type={buttons.type}
            callback={() => handleCallback()}
          ></Button>
        ) : null}
      </Column>
    </Card>
  );
};
