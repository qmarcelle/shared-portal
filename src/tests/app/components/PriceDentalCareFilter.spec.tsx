import { Filter } from '@/components/foundation/Filter';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(
    <Filter
      className="small-section px-0 m-0  md:w-[288px] w-auto"
      filterHeading="Filter Dental Costs"
      buttons={{
        className: 'font-bold',
        type: 'primary',
        label: 'Estimate Cost',
        callback: () => {},
      }}
      filterItems={[
        {
          type: 'dropdown',
          label: 'Dental Network',
          value: [
            {
              label: 'Choose Network',
              value: '1',
              id: '1',
            },
            {
              label: 'BlueCare Plus',
              value: '2',
              id: '2',
            },
            {
              label: 'BlueChoice (HMO)',
              value: '3',
              id: '3',
            },
            {
              label: 'In TN & border counties (DentalBlue)',
              value: '4',
              id: '4',
            },
            {
              label: 'Outside TN',
              value: '5',
              id: '5',
            },
          ],
          selectedValue: {
            label: 'Choose Network',
            value: '1',
            id: '1',
          },
        },
        {
          type: 'dropdown',
          label: 'Category',
          value: [
            {
              label: 'Select Category',
              value: '1',
              id: '1',
            },
            {
              label: 'Exams',
              value: '2',
              id: '2',
            },
            {
              label: 'Fluoride',
              value: '3',
              id: '3',
            },
            {
              label: 'Cleanings',
              value: '4',
              id: '4',
            },
            {
              label: 'X-rays',
              value: '5',
              id: '5',
            },
            {
              label: 'Sealants',
              value: '3',
              id: '3',
            },
            {
              label: 'Space Maintainer Bilateral',
              value: '6',
              id: '6',
            },
            {
              label: 'Fillings',
              value: '7',
              id: '7',
            },
            {
              label: 'Crowns and related services',
              value: '8',
              id: '8',
            },
            {
              label: 'Periodontal Surgery and related services ',
              value: '9',
              id: '9',
            },
            {
              label: 'Dentures and related services',
              value: '10',
              id: '10',
            },
            {
              label: 'Extractions-Simple and Surgical',
              value: '11',
              id: '11',
            },
            {
              label: 'Fixed Bridge',
              value: '12',
              id: '12',
            },
            {
              label: 'Endodontics',
              value: '13',
              id: '13',
            },
          ],
          selectedValue: {
            label: 'Select Category',
            value: '1',
            id: '1',
          },
        },
        {
          type: 'dropdown',
          label: 'Procedure',
          value: [
            {
              label: 'Select Procedure',
              value: '1',
              id: '1',
            },
            {
              label: 'Topical fluoride varnish',
              value: '2',
              id: '2',
            },
          ],
          selectedValue: {
            label: 'Select Procedure',
            value: '1',
            id: '1',
          },
        },
        { type: 'input', label: 'Zip Code', value: '' },
      ]}
    />,
  );
};

describe('PriceDentalCareFilter', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    expect(screen.getByText('Dental Network')).toBeVisible();
    expect(screen.getByText('Category')).toBeVisible();
    expect(screen.getAllByText('Procedure'));
    expect(screen.getByText('Zip Code')).toBeVisible();
    // expect(
    //   screen.getAllByText('Get started searching for a dentist near you.'),
    // );

    expect(component).toMatchSnapshot();
  });
});
