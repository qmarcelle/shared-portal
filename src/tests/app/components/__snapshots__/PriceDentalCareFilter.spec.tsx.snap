// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDentalCareFilter should render the UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-main small-section px-0 m-0  md:w-[288px] w-auto"
      >
        <div
          class="flex flex-col"
        >
          <h1
            class="title-1 title-2"
          >
            Filter Dental Costs
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <div
            class="flex flex-col"
          >
            <div
              class="body-1"
            >
              Dental Network
               
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
            />
            <div>
              <div
                class="rich-dropdown"
              >
                <div
                  class="flex flex-col switch-filter default"
                  tabindex="1"
                >
                  <div>
                    <div
                      class="body-1 input"
                    >
                      <div
                        class="flex flex-row p-1 items-center"
                      >
                        <div
                          class="flex flex-col border-none flex-grow"
                        >
                          <p
                            class="body-1"
                          >
                            Choose Network
                          </p>
                        </div>
                        <img
                          alt=""
                          class="w-[20px] h-[20px] ml-2 items-end"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="flex flex-col"
          >
            <div
              class="body-1"
            >
              Category
               
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
            />
            <div>
              <div
                class="rich-dropdown"
              >
                <div
                  class="flex flex-col switch-filter default"
                  tabindex="1"
                >
                  <div>
                    <div
                      class="body-1 input"
                    >
                      <div
                        class="flex flex-row p-1 items-center"
                      >
                        <div
                          class="flex flex-col border-none flex-grow"
                        >
                          <p
                            class="body-1"
                          >
                            Select Category
                          </p>
                        </div>
                        <img
                          alt=""
                          class="w-[20px] h-[20px] ml-2 items-end"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="flex flex-col"
          >
            <div
              class="body-1"
            >
              Procedure
               
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
            />
            <div>
              <div
                class="rich-dropdown"
              >
                <div
                  class="flex flex-col switch-filter default"
                  tabindex="1"
                >
                  <div>
                    <div
                      class="body-1 input"
                    >
                      <div
                        class="flex flex-row p-1 items-center"
                      >
                        <div
                          class="flex flex-col border-none flex-grow"
                        >
                          <p
                            class="body-1"
                          >
                            Select Procedure
                          </p>
                        </div>
                        <img
                          alt=""
                          class="w-[20px] h-[20px] ml-2 items-end"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="flex flex-col"
          >
            <span
              style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
            />
            <div
              class="flex flex-col w-full text-field"
            >
              <p>
                Zip Code
              </p>
              <div
                class="flex flex-row items-center input   "
              >
                <input
                  aria-label="Zip Code"
                  class=""
                  type="text"
                  value=""
                />
                <div
                  class="cursor-pointer"
                />
              </div>
              <div
                class="error-container mt-1 p-1"
              >
                <div />
              </div>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <button
            aria-label="Estimate Cost"
            class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null font-bold"
            type="button"
          >
            Estimate Cost
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-main small-section px-0 m-0  md:w-[288px] w-auto"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="title-1 title-2"
        >
          Filter Dental Costs
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="body-1"
          >
            Dental Network
             
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <div>
            <div
              class="rich-dropdown"
            >
              <div
                class="flex flex-col switch-filter default"
                tabindex="1"
              >
                <div>
                  <div
                    class="body-1 input"
                  >
                    <div
                      class="flex flex-row p-1 items-center"
                    >
                      <div
                        class="flex flex-col border-none flex-grow"
                      >
                        <p
                          class="body-1"
                        >
                          Choose Network
                        </p>
                      </div>
                      <img
                        alt=""
                        class="w-[20px] h-[20px] ml-2 items-end"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="body-1"
          >
            Category
             
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <div>
            <div
              class="rich-dropdown"
            >
              <div
                class="flex flex-col switch-filter default"
                tabindex="1"
              >
                <div>
                  <div
                    class="body-1 input"
                  >
                    <div
                      class="flex flex-row p-1 items-center"
                    >
                      <div
                        class="flex flex-col border-none flex-grow"
                      >
                        <p
                          class="body-1"
                        >
                          Select Category
                        </p>
                      </div>
                      <img
                        alt=""
                        class="w-[20px] h-[20px] ml-2 items-end"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="body-1"
          >
            Procedure
             
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <div>
            <div
              class="rich-dropdown"
            >
              <div
                class="flex flex-col switch-filter default"
                tabindex="1"
              >
                <div>
                  <div
                    class="body-1 input"
                  >
                    <div
                      class="flex flex-row p-1 items-center"
                    >
                      <div
                        class="flex flex-col border-none flex-grow"
                      >
                        <p
                          class="body-1"
                        >
                          Select Procedure
                        </p>
                      </div>
                      <img
                        alt=""
                        class="w-[20px] h-[20px] ml-2 items-end"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <div
            class="flex flex-col w-full text-field"
          >
            <p>
              Zip Code
            </p>
            <div
              class="flex flex-row items-center input   "
            >
              <input
                aria-label="Zip Code"
                class=""
                type="text"
                value=""
              />
              <div
                class="cursor-pointer"
              />
            </div>
            <div
              class="error-container mt-1 p-1"
            >
              <div />
            </div>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <button
          aria-label="Estimate Cost"
          class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null font-bold"
          type="button"
        >
          Estimate Cost
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
